src/amo/components/AddonReviewCard/index.js

```diff
@@ -405,6 +402,43 @@ export class AddonReviewCardBase extends React.Component<InternalProps> {
         ],
       });
 
+      const byAuthorOnly = i18n.sprintf(i18n.gettext('by %(authorName)s'), {
+        authorName: review.userName,
+      });
+
+      const byAuthorLink = replaceStringsWithJSX({
+        text: i18n.sprintf(
+          i18n.gettext('by %(urlStart)s%(authorName)s%(urlEnd)s'),
+          {
+            authorName: review.userName,
+            urlStart: '%(urlStart)s',
+            urlEnd: '%(urlEnd)s',
+          },
+        ),
+        replacements: [
+          [
+            'urlStart',
+            'urlEnd',
+            (text) => (
+              <Link key={review.userName} to={`/user/${review.userId}`}>
+                <span>{text}</span>,{' '}
+              </Link>
+            ),
+          ],
+        ],
+      });
+
+      const byAuthor = !review.userUrl ? byAuthorOnly : byAuthorLink;
+
+      const byLineLink = noAuthor ? (
+        byLineTimestamp
+      ) : (
+        <>
+          <span>{byAuthor}</span>
+          <span>{byLineTimestamp}</span>
+        </>
+      );
+
```

Comment:
I think we want something a bit different. I guess there should be 3 different translation strings for `byLineString`:

- when we have no review (current behavior)
- when we have a review and the current logged in user is not an admin (current behavior)
- when we have a review and the current logged in user is an admin (new!)

Then we'll have to update the code near `replaceStringsWithJSX()`, which is not very friendly. For each `<Link />`, this function needs:

- two placeholders that will be replaced by the same values (one for "start" and the other for "end"). I suggest we name these placeholders: `linkUserProfileStart` and `linkUserProfileEnd`. These placeholders are similar to `linkEnd` and `linkStart` in `byLineString`
- a new "replacement entry" in the `replacements` list (still in `replaceStringsWithJSX()` but a different key): This array is an array of arrays, and each array contains 3 parameters: the start placeholder, the end placeholder and a function.

I think the placeholders replaced by themselves should be `undefined` when there is no user OR the user is NOT an admin, too (so it is not just `linkUserProfileEnd: '%(linkUserProfileEnd)s'` but maybe something like `linkUserProfileEnd: !noAuthor && hasAdminPermission && '%(linkUserProfileEnd)s'` but that's just an example).

Next, we only want a new replacement entry if there is a user AND the user is an admin so we'll have to extract the `replacements` array in the `replaceStringsWithJSX()` call in a new variable (above the call to `replaceStringsWithJSX()`) and `.push()` a new replacement entry if the user is an admin.

After that, we can construct the `<Link />` in the function (third parameter of a replacement entry). This is straightforward because we only have to return a `<Link />` all the time, pointing to `/user/${review.userId}/`. We'll see about passing the `clientApp`/`lang` later (not sure that's needed).

Once we have that, we want to write new test cases (and make sure existing ones don't fail because they shouldn't).

Reply:
I did not realize that I could add conditions to the placeholders! This was extremely helpful, thank you!!!!  🎉